### DISCUSSION TITLE
Use of constructor in Form Class

### DISCUSSION THREAD

**Discussion by Anglin-George** [livewire/livewire#7163] [1. Help] [answered]
I created a form class that extends the create & edit livewire page for package manage. I have a common Currency model and for that purpose in call it in form constructor. It's worked nicely but there showing a strange behavior on showing error message.

`

    #[Rule('required')]
    public $overview;

    #[Rule('required')]
    public $description;

    #[Rule(['form.images.*' => 'nullable|sometimes|image'])]
    public $images = [];

    #[Rule(['form.price.currency_id' => 'required|numeric', 'form.price.adult_amount' => 'required|numeric|min:0|max:10000000', 'form.price.children_amount' => 'required|numeric|min:0|max:10000000'])]
    public $price = [];   

    public $uploadedImages = [];
    public $currencies = [];

    public function __construct()
    {
        $this->fill([
            'currencies' => Currency::all(),
        ]);
    }
    
    public function messages() 
    {
        return [
            'name.required' => 'Please type package :attribute',
            'overview.required' => 'Please type :attribute',
            'description.required' => 'Please type :attribute',
            'images.*.image' => 'Please upload valid image',
            'price.currency_id.required' => 'Please select currency',
            'price.adult_amount.required' => 'Please type amount',
            'price.adult_amount.min' => 'Please type valid amount',
            'price.children_amount.required' => 'Please type amount',
            'price.children_amount.min' => 'Please type valid amount',
        ]; 
    }  

` 
Below I posted the error message. When I removed the constructor function the validation shows correct. Any work arounds or I'm I doing something wrong here ?

![image](https://github.com/livewire/livewire/assets/30459349/16401ee3-3182-4ab7-8351-12b5aa8ce674)

**Comment by satoved** [comment]
You've overwritten the form constructor. Try calling the parent constructor first.

**Reply by Anglin-George** [reply]
Thank you @satoved for your response.
If there is a method exist to inject these types of functions to constructor will be more useful like mount in Livewire components.

**Comment by joshhanley** [comment, marked as answer]
@Anglin-George just like Livewire components, you shouldn't use the constructor. Just make a method on your FormObject and call that from mount or boot in your Livewire component to initialise it.

**Reply by Anglin-George** [reply]
Hi @joshhanley ,
Yes, I solved the issue like this and If there is a method like mount in FormObject will be very helpful.  

Thanks in advance ☺️
